arch/Kconfig

```diff
@@ -29,6 +29,8 @@ config ARM
 	bool
 	select ARCH_IS_SET
 	select ARCH_SUPPORTS_COREDUMP if CPU_CORTEX_M
+	# GDBSTUB has not yet been tested on Cortex M or R SoCs
+	select ARCH_HAS_GDBSTUB if CPU_AARCH32_CORTEX_A
```

Comment:
Maybe put this inside `arch/arm/core/aarch32/Kconfig` under `CPU_AARCH32_CORTEX_A`? M and R can do the same to their kconfigs when support is added.